readme.md

```diff
@@ -124,7 +124,7 @@ Default: `{}`
 
 Request headers.
 
-Existing headers will be overwritten. Headers set to `null` or `undefined` will be omitted.
```

Comment:
I've removed that part to not confuse people with the behaviour of `got.mergeOptions`, where setting to `undefined` may keep the old value.